### ISSUE TITLE
[Spike]Research Rook Operator Ceph cluster backup and restore

### ISSUE THREAD

**Issue by mkyc** [epiphany-platform/epiphany#1451]
**Is your feature request related to a problem? Please describe.**
We don't have any information about backup and restore procedure for Rook Ceph Operator.

**Describe the solution you'd like**
We should check how to handle backup/restore of Ceph Cluster (data).

**Describe alternatives you've considered**
No alternatives.

**Additional context**
No additional context.

**Comment by rpudlowski93** [comment]
In case of rook ceph cluster will crash, we would like to have data backup of our application/database which use pvc, completely independent from the ceph cluster. After a deep research and tests, I have to confirm that it is **no possible** to do it using [volume snapshots](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) in kubernetes. 

Rook Ceph support only two CSI Drivers: RBD CSI driver and CephFS CSI driver. Please note that the source PVC, PV, and VolumeSnapshotClass for a VolumeSnapshot object must point to **the same CSI driver**. I was trying to test it and find any opportunity to do it but without any success. 

Volume snapshots works great using for example AzureDisk CSI for "plain" PVs and snapshots as well but in our case when I would like to do it with Rook Ceph Cluster and  PVc created with RBD/CephFS CSI Driver, error appears:  `GRPC error: rpc error: code = InvalidArgument desc = could not get resource group from diskURI(0001-0009-rook-ceph-0000000000000001-14dc1b34-e7cc-11ea-891b-6ac1584e0dfd) with error(invalid disk URI: 0001-0009-rook-ceph-0000000000000001-14dc1b34-e7cc-11ea-891b-6ac1584e0dfd)`

Summing up, in order to make a data backup in rook ceph , we can not use for that volume snapshots. It is no possible right now. There is one rook ceph backup tool which we should take into account [link](https://gitlab.com/jrevolt/rook-ceph-backup). There is always possible to make a snapshot on the cloud provider level of the entire data disk, need consider.

**Comment by ar3ndt** [comment]
Backup/Restore is possible using RBD kind of storageclass that PVC uses.
Once this is satisfied we need to do the following steps:

- Create toolbox pod with rbd tools installed (example yaml file it is located in rook repo examples)
- Make some changes on the pod you want to backup (within the mounted pvc location)
- Connect `rook-ceph-tools` container
- Check poolname that was created for the storage class that your PVC uses.
  `rados lspools`
- Check volume name
  `rbd ls replicapool`
- Create snapshot
   `rbd snap create replicapool/csi-vol-365399b7-e79e-11ea-a8cc-66630cbb1481@test1-snapshot`
- Make some more changes in the pod. 
- Scale down application
`kubectl scale --replicas=0 deployment/POD_NAME`
- Rollback from snapshot (again within toolbox container)
`rbd snap rollback replicapool/csi-vol-365399b7-e79e-11ea-a8cc-66630cbb1481@test1-snapshot`
- Expected result:
```
1ea-a8cc-66630cbb1481@test1-snapshot
Rolling back to snapshot: 100% complete...done.
```
- Scale up application
`kubectl scale --replicas=1 deployment/POD_NAME`

Changes should be rolled back

**Comment by ar3ndt** [comment]
Export / Import tested and seems to be working fine.
Created ceph cluster on 2 separate environments: AWS and AZURE.

Exported changed data in postgresql pod using PVC with Ceph block storage class.
Imported the data on the other cluster and observed successfully changed data on the other cluster.

**Comment by rpudlowski93** [comment]
Nothin to do here more, tomorrow demo will be presented.
